core/src/main/protobuf/org/apache/spark/status/protobuf/store_types.proto

```diff
@@ -765,3 +765,54 @@ message PoolData {
   optional string name = 1;
   repeated int64 stage_ids = 2;
 }
+
+message StateOperatorProgress {
+  optional string operator_name = 1;
+  int64 num_rows_total = 2;
+  int64 num_rows_updated = 3;
+  int64 all_updates_time_ms = 4;
+  int64 num_rows_removed = 5;
+  int64 all_removals_time_ms = 6;
+  int64 commit_time_ms = 7;
+  int64 memory_used_bytes = 8;
+  int64 num_rows_dropped_by_watermark = 9;
+  int64 num_shuffle_partitions = 10;
+  int64 num_state_store_instances = 11;
+  map<string, int64> custom_metrics = 12;
```

Comment:
I am concerned about the nullability of all these maps. Shall we check/test all of them?

Reply:
[a904a27](https://github.com/apache/spark/pull/39642/commits/a904a27919a47cebf3784a8756f46b3237b4be46) check/test all map 

[699ebd1](https://github.com/apache/spark/pull/39642/commits/699ebd1e6c3905722d0b09ff11e5dccc31813d3c) add  `setJMapField`  function to `Utils`